fmriprep/workflows/confounds.py

```diff
@@ -243,3 +262,197 @@ def reverse_order(inlist):
     if isinstance(inlist, list):
         inlist.reverse()
     return inlist
+
+
+def get_ica_confounds(ica_out_dir):
+    import os
+    import numpy as np
+    from niworkflows.nipype import logging
+
+    # To catch edge cases when there are either no noise or signal components
+    LOGGER = logging.getLogger('workflow')
+
+
+    # Pass in numpy array and column base name to generate headers
+    # modified from add_header_func
+    def aroma_add_header_func(np_arr, col_base, comp_nums):
+        import pandas as pd
+        from sys import version_info
+        PY3 = version_info[0] > 2
+
+        df = pd.DataFrame(np_arr, columns=[str(col_base) + str(index) for index in comp_nums])
+        df.to_csv(str(col_base) + "ica_confounds.tsv",
+                  sep="\t" if PY3 else '\t'.encode(), index=None)
+
+        return os.path.abspath(str(col_base) + "ica_confounds.tsv")
+
+    # Partial regression (close to, but not identical to fsl_regfilt)
+    def calc_residuals(x, y):
+        X = np.column_stack(x + [[0] * len(x[0])])
```

Comment:
That makes more sense to me. Adding a column of ones calculates a mean regressor (the `c` in their example, so your residuals will be de-meaned. Which is probably what you want, but I just want to be explicit about this, in case that's not actually what you want, here.